packages/browser/src/skylabClient.ts

```diff
@@ -106,43 +130,126 @@ export class SkylabClient implements Client {
+      throw e;
+    }
+  }
+
+  protected async doFetch(
+    user: SkylabUser,
+    timeoutMillis: number,
+  ): Promise<Response> {
+    const userContext = this.addContext(user);
+    const encodedContext = urlSafeBase64Encode(JSON.stringify(userContext));
+    let queryString = '';
+    let debugAssignmentRequestsParam: string;
+    if (this.debugAssignmentRequests) {
+      debugAssignmentRequestsParam = `d=${randomString(8)}`;
+    }
+    if (debugAssignmentRequestsParam) {
+      queryString = '?' + debugAssignmentRequestsParam;
+    }
+    const endpoint = `${this.config.serverUrl}/sdk/vardata/${encodedContext}${queryString}`;
+    const headers = {
+      Authorization: `Api-Key ${this.apiKey}`,
+    };
+    if (this.debug) {
+      console.debug('[Skylab] Fetch variants for user: ', userContext);
+    }
+    const response = await this.httpClient.request(
+      endpoint,
+      'GET',
+      headers,
+      null,
+      timeoutMillis,
+    );
+    if (this.debug) {
+      console.debug('[Skylab] Received fetch response:', response);
+    }
+    return response;
+  }
+
+  protected async parseResponse(response: Response): Promise<Variants> {
+    const json = await response.json();
+    const variants: Variants = {};
+    for (const flag of Object.keys(json)) {
+      variants[flag] = {
+        value: json[flag].key,
+        payload: json[flag].payload,
       };
-      const response = await this.httpClient.request(endpoint, 'GET', headers);
-      const json = await response.json();
-      this.storage.clear();
-      for (const flag of Object.keys(json)) {
-        this.storage.put(flag, {
-          value: json[flag].key,
-          payload: json[flag].payload,
-        });
-      }
-      this.storage.save();
+    }
+    if (this.debug) {
+      console.debug('[Skylab] Received variants:', variants);
+    }
+    return variants;
+  }
+
+  protected storeVariants(variants: Variants): void {
+    this.storage.clear();
+    for (const key in variants) {
+      this.storage.put(key, variants[key]);
+    }
+    this.storage.save();
+    if (this.debug) {
+      console.debug('[Skylab] Stored flags:', variants);
+    }
+  }
 
+  protected startRetries(): void {
+    if (this.debug) {
+      console.debug('[Skylab] Retry fetch all');
+    }
+    // Non-zero retry means we already have a retry interval in progress.
+    if (this.retries) {
       if (this.debug) {
-        console.debug('[Skylab] Received and stored flags:', json);
+        console.debug('[Skylab] Retry fetch interval already in progress');
       }
-    } catch (e) {
-      console.error(e);
+      return;
     }
-    return this;
+    // Run fetchAll for the current user at an interval defined by the
+    // fetchRetryIntervalMillis config.
+    this.retries = globalThis.setInterval(async () => {
+      try {
+        await this.fetchAll(
+          this.user,
+          this.config.fetchRetryTimeoutMillis,
+          false,
+        );
+        this.stopRetries();
```

Comment:
It looks like we're only retrying once and then clearing the interval, is that by design? If so, I would rename `this.retries` to `this.retryHandle`

Reply:
`fetchAll` will throw on error, so `stopRetries()` should only get called on success.

Reply:
Ah if that's the case, will this just retry forever? I'd put in a (configurable) retry limit if that's the case

Reply:
Yes, that is currently the case. I figured we would want to retry as much as possible until we can get variants for the user to cache in local storage. If you look at skylab-js-server PR I could do something similar (e.g. configurable max number of retries with exponential backoff.